study1/week5/group-anagrams.java

```diff
@@ -0,0 +1,41 @@
+// https://leetcode.com/problems/group-anagrams/
+// Time Complexity: O(nlogn), n: length of string which is the longest
+// Space Complexity: O(m+n), m: strMap.size, n: anagramList.size
+class Solution {
+    public List<List<String>> groupAnagrams(String[] strs) {
+        HashMap<String, ArrayList<String>> strMap = new HashMap<String, ArrayList<String>>();    // key: sorted String, value: String list
+        List<List<String>> anagramList = new ArrayList<List<String>>();
+
+        for (String str : strs) {
+            String key = sortString(str);
+            if (strMap.containsKey(key)) {
+                ArrayList<String> strList = strMap.get(key);
+                strList.add(str);
+                strMap.put(key, strList);
```

Comment:
You don't have to put `strList` into the map because `strMap.get(key) == strList` already